src/drt/src/pa/FlexPA_acc_pattern.cpp

```diff
@@ -778,8 +783,8 @@ bool FlexPA::genPatternsCommit(
   if (target_obj != nullptr
       && genPatternsGC({target_obj}, objs, Commit, &owners)) {
     pin_access_pattern->updateCost();
-    const int inst_idx = unique_insts_.getIndex(inst);
-    unique_inst_patterns_[inst_idx].push_back(std::move(pin_access_pattern));
+#pragma omp critical
+    unique_inst_patterns_[unique_inst].push_back(std::move(pin_access_pattern));
```

Comment:
Not critical. No other thread would access the same entry for the current unique_inst. And we already created the entry beforehand, so the map would not be touched.